internal/publish/model/stats.go

```diff
@@ -57,6 +61,60 @@ type HealthAuthorityStats struct {
 	MissingOnset      int64
 }
 
+// ReduceStats takes hourly breakdowns and rolls them up to daily. The onlyBefore
+// time indicates the cutoff point for inclusion.
+// The dayThreshold indicates how many entries are needed to include a given day.
+// Day boundaries are all in UTC.
+func ReduceStats(hourly []*HealthAuthorityStats, onlyBefore time.Time, dayThreshold int64) []*verifyapi.MetricsDay {
+	days := make(map[time.Time]*verifyapi.MetricsDay)
+	// Combine the hours into days based on same UTC midnight time.
+	for _, hour := range hourly {
+		if !hour.Hour.Before(onlyBefore) {
+			continue
+		}
+
+		day := timeutils.UTCMidnight(hour.Hour)
+		if _, ok := days[day]; !ok {
+			// Initialize this day
+			days[day] = &verifyapi.MetricsDay{
+				Day:                   day,
+				OldestTEKDistribution: make([]int64, StatsMaxOldestTEK+1),
```

Comment:
These +1 feel arbitrary?

Reply:
it's not - the +1 is the > max bucket throughout